workflow/controller/operator.go

```diff
@@ -140,6 +143,19 @@ type failedNodeStatus struct {
 
 // newWorkflowOperationCtx creates and initializes a new wfOperationCtx object.
 func newWorkflowOperationCtx(wf *wfv1.Workflow, wfc *WorkflowController) *wfOperationCtx {
+	var rateLimiter *waitutil.RateLimiter = nil
+
+	intervalMsStr, present := os.LookupEnv(common.EnvCleanupRateLimitIntervalMilliSeconds)
+	if present {
+		intervalMs, err := strconv.Atoi(intervalMsStr)
+		if err != nil {
+			log.Errorf("was not able to convert %s to int, defaulting to queue based cleanup, err: %s", intervalMsStr, err)
+		} else {
+			rl := waitutil.NewRateLimiter(time.Duration(intervalMs) * time.Millisecond)
+			rateLimiter = &rl
+		}
+	}
```

Comment:
the rate limiter is just being used for batch deletion, right? Can we add a comment to that effect here?

Reply:
Sure will do!